src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala

```diff
@@ -60,15 +74,43 @@ class TileUtils {
     column.toFloat / math.pow(2, zoom) * 360.0 - 180.0
   }
 
+  def tile_from_tile_id_str(tileId: String): Tile = {
+    tile_from_tile_id(new TileId(tileId))
+  }
+
   def tile_from_tile_id(tileId: TileId): Tile = {
     new Tile(tileId, latitude_from_row(tileId.row, tileId.zoom),
       latitude_from_row(tileId.row + 1, tileId.zoom), longitude_from_column(tileId.column, tileId.zoom),
       longitude_from_column(tileId.column + 1, tileId.zoom))
   }
 
   def tile_id_from_row_column(row: Int, column: Int, zoom: Int ): TileId = {
-    val tileStr = printf("%d_%d_%d", zoom, row, column)
-    new TileId(tileStr.toString)
+    new TileId(f"$zoom%d_$row%d_$column%d")
+  }
+
+  def map_to_zoom_render_tile(tileTuple: (String, Int)): (String, (String, Int)) = {
+    val tile = TileUtils().tile_from_tile_id_str(tileTuple._1)
+    //return the tile id 5 levels deep from tileTuple which will be used to render in the heatmap
+    (TileUtils().tile_id_from_lat_long(tile.centerLatitude(), tile.centerLongitude(), tile.tileId.zoom - DETAIL_ZOOM_DELTA).tileId,
```

Comment:
Why do we need to instantiate the `TileUtils` here? Should we just turn the class into a `object` so that we can access the helpers statically?